faunadb-scala/src/main/scala/faunadb/streaming/BodyValueFlowProcessor.scala

```diff
@@ -0,0 +1,83 @@
+package faunadb.streaming
+
+import java.nio.ByteBuffer
+import java.util
+import java.util.concurrent.{Flow, SubmissionPublisher}
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.{JsonMappingException, ObjectMapper}
+import faunadb.errors.StreamingException
+import faunadb.values.{StringV, VSuccess, Value}
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConverters.asScalaIteratorConverter
+
+class BodyValueFlowProcessor extends SubmissionPublisher[Value] with Flow.Processor[util.List[ByteBuffer], Value] {
+  private val log = LoggerFactory.getLogger(getClass)
+  private val json = new ObjectMapper
+  private var subscription: java.util.concurrent.Flow.Subscription = null
+  private var subscriber: Flow.Subscriber[_ >: Value] = null
+
+  // We do not request data from the publisher until we have one subscriber
+  override def subscribe(subscriber: Flow.Subscriber[_ >: Value]): Unit = {
+    if (this.subscriber == null) {
+      this.subscriber = subscriber
+      super.subscribe(subscriber)
+      requestOne()
+    } else
+      throw new IllegalStateException("BodyValueFlowProcessor can have only one subscriber")
+  }
+
+  override def onSubscribe(subscription: Flow.Subscription): Unit =
+    this.subscription = subscription
+
+  override def onNext(item: util.List[ByteBuffer]): Unit = {
+    import java.nio.charset.StandardCharsets
+    val text = item.iterator().asScala.map(StandardCharsets.US_ASCII.decode(_).toString).mkString("")
+    try {
```

Comment:
Could we use the monad `Try` here instead? :)

Reply:
In general I don't use the `Try` when there is no monadic composition nor when the type is exposed to the caller.
If you feel strongly about it I can change it.
WDYT?